starter_project/autonomy/src/perception.cpp

```diff
@@ -71,12 +123,28 @@ namespace mrover {
         // hint: try not overthink, this metric does not have to be perfectly accurate, just correlated to distance away from a tag
 
         // TODO: implement me!
-        return {};
+        //return percent of screen taken off by corners
+        //assume the tag is a square?
+        int total_area = image.rows * image.cols;
+        
+        float corner_area = abs((tagCorners[1].x - tagCorners[0].x) * (tagCorners[0].y - tagCorners[3].y));
+
+        return 1 - float((1.0*corner_area) / (1.0*total_area));
```

Comment:
Pull out `float((1.0*corner_area) / (1.0*total_area));` as a variable for readability